packages/fxa-settings/.rescriptsrc.js

```diff
@@ -1,11 +1,35 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const { resolve } = require('path');
+
 const {
   permitAdditionalJSImports,
   setupAliasedPaths,
-  componentsJestMapper
-} = require("../fxa-components/rescripts");
+  componentsJestMapper,
+} = require('../fxa-components/rescripts');
+
+function addSvgLoader(config) {
+  config.module.rules.push({
+    test: /\.svg$/,
+    include: resolve('./src'),
+    use: [
+      {
+        loader: 'babel-loader',
+      },
+      {
+        loader: 'react-svg-loader',
```

Comment:
Hm. No, this wouldn't work for a `background-image` URL, as this loads the actual SVG, and not any sort of encoded version or URL. I can switch to that style of loader, but at some point I'd like to be able to load actual SVG files in (and not just a URL).

Regarding, aria-label, this approach allows you to supply custom props, so you could set an aria-label, but I don't think that should be a concern of the loader, and is more just a thing we need to remember to do. Maybe we can lint it.

Reply:
> is more just a thing we need to remember to do

Yeah, this is what I hope to avoid, because we will forget. 😂 Images without alt tags will complain in the console but these won't so it's a bit trickier. I'm just wondering if there's a way to require it through TS or yeah maybe some linting option. I'm still wondering if `svg-url-loader` would be the better option though since it would also inline our `background-image`s (and we will have at least a few) and we could use `img` and not forget `alt`. What's the benefit of importing them as React components if we can set `width`/`height` properties in CSS on an image? FWIW I like the idea of `react-svg-loader` better but wondering if the other would serve our purposes a bit better?

Reply:
> What's the benefit of importing them as React components if we can set width/height properties in CSS on an image?

When you use an SVG as an image src you lose the ability to manipulate the SVG itself, via JS or CSS. For example, if we wanted to set the fill or stroke of a button icon on hover, or maybe we want apply an animation to one of the groups in an SVG, or maybe we'd like to target an SVG node with JavaScript to dynamically update it's path.

_That being said..._ I don't envision a lot of that happening with this project. Maybe the camera icon for the cropper, but that's probably it. I'll change it to a URL/encoded loader, and we can address stuff as it comes up.

Reply:
Good news: CRA handles **all of the SVG loading**. I'll reflect this in the Readme with examples :)